### ISSUE TITLE
Allow option to ignore "OpenClipboard failed" for "Copy Link to Page", when Clipboard Format Spy is running

### ISSUE THREAD

**Issue by jabobian** [stevencohn/OneMore#1071]
# Problem to Solve

When Click Reference >> Copy Link to Page, some times it will show a warning dialog saying "Error running Command CopyLinkCommand ...", the log shows

```
13| Error running CopyLinkCommand
13| System.Runtime.InteropServices.COMException
Message: OpenClipboard Fails (Exception from HRESULT:0x800401D0 (CLIPBRD_E_CANT_OPEN))
Description: Unrecognized error code 0x800401D0
ErrorCode: 0x800401D0 (-2147221040)
HResult: 0x800401D0 (-2147221040)
Source: mscorlib
StackTrace:    在 System.Runtime.InteropServices.Marshal.ThrowExceptionForHRInternal(Int32 errorCode, IntPtr errorInfo)
  At System.Windows.Clipboard.Flush()
```

### **But the clipboard is indeed filled with correct link to page ** 

# To Reproduce

1. Install DelphiDabbler Clipboard Format Spy ([https://github.com/delphidabbler/cfs](https://github.com/delphidabbler/cfs)) and run CFS.exe
2. Go to 'Reference'
3. Click on 'Copy Link to Page'

# Expected Behavior

# Screenshots

# Environment (if applicable)

 - OneNote Version: Office 2016, 64bit
 - OneMore Version 5.8.5

# Additional Context

Add any other context about the problem here. Attach page XML file or the app log file as appropriate. You can find the log file here: %localappdata%\temp\OneMore.log

THE FINE PRINT This is a hobby project that I started for myself. I'm happy to take requests and will promise to
consider each one carefully but offer no guarantees that I will ultimately agree to their
usefulness or have time to implement any of them. If OneNote offers a "close enough" work-around
then I will probably reject the request - you've been warned!

**Comment by jasonjac2** [comment]
I have the same thing with a clipboard monitor I use, i.e. the clipboard error.  I am say 65 - 75% convinced that OneMore attempts to lock the clipboard twice in close succession, although in #995 @stevencohn checked and didn't think it did. I don't think ignoring it is the right thing to do, I feel like OM should try and try a couple more times.

**Comment by stevencohn** [comment]
Given the example above using DelphiDabbler, I now have a use case that I may be able to replicate. I will give it a try.

**Comment by jasonjac2** [comment]
Aside: What a small world - DelphiDabbler = an UK Embarcadero MVP like myself - but I think he has retired.  I was hoping that someone who uses OneMore also uses Delphi, but No :-(.  I am struggling to find anyone using Delphi who is developing against oneNote

**Comment by jasonjac2** [comment]
Hi @stevencohn I don't think this problem is fixed, but just masked.  If you remember I watch the clipboard in one of my applications (Project Filter) and reported a while aga.  Anyway, you have this fixed in 5.9.0, but it looks like you just swallow the exception. So if my little app is running and I use "copy link to paragrah", I don't get an error, but not all formats are copied onto the clipboard.
Basically if I paste the link back into ON, I get the full onenote:............ type link where the text is what you would get if you pasted into notepad., e.g. onenote:https://d.docs.live.net/<snip>&section-id=<snip>  etc.
Instead of the text: notebook -> section -> page -> text (all as one link).

I hope this makes sense.

**Comment by stevencohn** [comment]
You're getting the correct content but it's not formatted correctly? That would imply that some but not all formats are available on the clipboard so it's still a locking/timing issue.

If you have a scenario that is 100% reproducible, please describe it so I can replicate and diagnose.

**Comment by jasonjac2** [comment]
e-mailed you.

**Comment by jasonjac2** [comment]
@stevencohn I have moved onto a temporary laptop and I am getting this 95% of the time. I think the tools that interfere are *ANY* of the following running: Powetoys; RDPing onto another PC (with clipboard hook) VNCing onto another PC.  I am getting the exception in the log, running current (6.4.0).  I use the copy link to page and copy link to para probably about 10 times an hour, and they all (when pasted into ON) produce a link that is: onenote:https://d.docs.live.net...........
Log extract:
Message: OpenClipboard Failed (Exception from HRESULT: 0x800401D0 (CLIPBRD_E_CANT_OPEN))
Description: Unrecognized error code 0x800401D0
ErrorCode: 0x800401D0 (-2147221040)
HResult: 0x800401D0 (-2147221040)
Source: mscorlib
StackTrace:    at System.Runtime.InteropServices.Marshal.ThrowExceptionForHRInternal(Int32 errorCode, IntPtr errorInfo)
   at System.Windows.Clipboard.Flush()
   at System.Windows.Clipboard.CriticalSetDataObject(Object data, Boolean copy)
   at River.OneMoreAddIn.ClipboardProvider.<>c__DisplayClass14_0.<RestoreState>b__0()
TargetSite: [mscorlib] System.Runtime.InteropServices.Marshal::ThrowExceptionForHRInternal()

**Comment by jasonjac2** [comment]
So it feels like keyboard failures are common and other apps just kind of deal with them, it feels like the old NW congestion and timing your retries.... 
So it looks like you grab the clipboard, set the plain text, then go back to set the other formats (HTML I am assuming), you aren't directly trying to set ONeNote2016 format directly.  This has all come to a head as my primary laptop started smoking and so I am moving between machines.

**Comment by jasonjac2** [comment]
Happy to create a new ticket if you would prefer.

**Comment by stevencohn** [comment]
Please create a new ticket.
